packages/app-js/src/index.tsx

```diff
@@ -3,57 +3,83 @@
 // of the Apache-2.0 license. See the LICENSE file for details.
 
 import { ApiProps } from '@polkadot/ui-api/types';
-import { AppProps, I18nProps } from '@polkadot/ui-app/types';
-import { Log, LogType } from './types';
+import { AppProps } from '@polkadot/ui-app/types';
+import { CustomWindow,Log, LogType } from './types';
 
 import React from 'react';
 import { withApi, withMulti } from '@polkadot/ui-api/index';
-import { Button } from '@polkadot/ui-app/index';
+import { Button, Dropdown } from '@polkadot/ui-app/index';
 import uiKeyring from '@polkadot/ui-keyring';
 import * as util from '@polkadot/util';
 import * as hashing from '@polkadot/util-crypto';
 
 import './index.css';
+import snippets from './snippets';
+import translate from './translate';
 
+import Intro from './Intro';
 import Editor from './Editor';
 import Output from './Output';
-import translate from './translate';
 
-type Props = ApiProps & AppProps & I18nProps;
+type Props = ApiProps & AppProps;
 type State = {
   code: string,
-  logs: Array<Log>
+  logs: Array<Log>,
+  snippet: string
 };
 
+const customWindow: CustomWindow = window;
+
+const local = snippets.find(obj => obj.value === localStorage.getItem('app-js-snippet'));
+
 class App extends React.PureComponent<Props, State> {
   state: State = {
-    code: '',
-    logs: []
+    code: local ? local.code : snippets[0].code,
+    logs: [],
+    snippet: local ? local.value : snippets[0].value
   };
 
   render () {
-    const { logs } = this.state;
+    customWindow.api = this.props.api;
```

Comment:
I'm still wondering about the exposure the API and keyring on window since it means that anything has access to it, including extensions and scripts. I'm still not convinced on it, it opens up potential avenues of attack/inspection without real additional benefit. 

There are benefits in fooling about, but doing this by default when we don't have to, I'm not sure about. (Been giving it some thought after it come up)

Reply:
Yeah, that's one thing I wanted to ask. It's also the reason why I only implemented it in here. 
It's really convenient to explore the api without having to look at the docs, though. I was thinking that we could maybe just expose it when connected to a dev chain?

Reply:
Happy with dev-only exposure. However we should expose it via the API itself then instead of the js part, probably around here - https://github.com/polkadot-js/apps/blob/master/packages/ui-api/src/Api.tsx#L92

So expose once initialised and we are on dev chain (we should be able to use this https://github.com/polkadot-js/apps/blob/master/packages/ui-api/src/util/isTestChain.ts - looking at it now, now 100% on the `dev|loc` match, explicit via `local` and `development` is probably better. I know why it is short, dev chains originally retuned `dev` instead of `development`)

Reply:
I also think the API can be a bit more explorable by default and without tools, i.e. if we add something like

```js
api.tx.listAll() // shows all section.methods with description
```

and apply it consistently on `tx`, `query`, `derive` (this one is probably tougher) and `rpc`, we also have a way of getting info for non-nice-editor enabled people. (Without actually adding bloat)

Outside the scope of this, but may be worth a thought... Another one which apps and Edgeware both seems to use is a `api.tx[method][section]` pattern where section + method is selectable by the user, here something like `api.tx.call(section, method)` can actually be cleaner. Also out of scope here... but since I started mucking in `api.tx`, thought I may as well share what I wondered about...

Reply:
> probably around here - https://github.com/polkadot-js/apps/blob/master/packages/ui-api/src/Api.tsx#L92

Good idea, I can have a look at it once this PR is merged.

Reply:
> ```js
> api.tx.listAll() // shows all section.methods with description
> ```

I like that! But that would also go directly in the api and not in here, right?

Reply:
100%, directly on the API.

Reply:
I put it down on my list to create tickets in /api for that.

Reply:
Note: I removed it